docs/how-tos/hello-world-module.md

```diff
@@ -0,0 +1,304 @@
+---
+title: "Create a Hello World module"
+linkTitle: "Hello World module"
+type: "docs"
+weight: 24
+images: ["/registry/module-puzzle-piece.svg"]
+icon: true
+tags: ["modular resources", "components", "services", "registry"]
+description: "Get started with custom module creation by creating a Hello World modular resource."
+languages: ["python", "go"]
+viamresources: ["components"]
+platformarea: ["registry"]
+level: "Beginner"
+date: "2024-10-22"
+# updated: ""  # When the tutorial was last entirely checked
+cost: "0"
+---
+
+This guide will walk you through creating a simple {{< glossary_tooltip term_id="modular-resource" text="modular resource" >}} that responds to API calls by returning a random number, and a configured image.
+The goal is to familiarize you with how to create your own {{< glossary_tooltip term_id="module" text="modules" >}} and [modular resources](/registry/modular-resources/).
+
+{{% alert title="In this page" color="tip" %}}
+
+1. [Concepts](#what-is-a-module-what-is-a-modular-resource)
+1. [Write the basic functionality into a test script](#write-the-basic-functionality-into-a-test-script)
+1. [Choose an API](#choose-an-api-to-implement)
+1. [Generate code stub files](#generate-stub-files)
+1. [Implement the API](#implement-the-api)
+1. [Test your module](#test-your-module)
+1. [Package the module](#package-the-module)
+
+{{% /alert %}}
+
+## Prerequisites
+
+{{< expand "Install the Viam CLI and authenticate" >}}
+Install the Viam CLI and authenticate to Viam, from the same machine that you intend to upload your module from.
+
+{{< readfile "/static/include/how-to/install-cli.md" >}}
+
+Authenticate your CLI session with Viam using one of the following options:
+
+{{< readfile "/static/include/how-to/auth-cli.md" >}}
+{{< /expand >}}
+
+{{% expand "viam-server installed on your computer and connected to the Viam app." %}}
+
+{{% snippet "setup.md" %}}
+
+{{% /expand%}}
+
+## What is a module? What is a modular resource?
+
+A module is a set of files that provides support for one or more {{< glossary_tooltip term_id="component" text="components" >}} or {{< glossary_tooltip term_id="service" text="services" >}} that are not built into `viam-server`.
+The {{< glossary_tooltip term_id="resource" text="resources" >}} supported by a module are called modular resources.
+
+## Write the basic functionality into a test script
```

Comment:
Shorten (also you use a lot of "simple" and "basic") - I left the simple above but I think the basic is unnecessary. 
```suggestion
## Create a test script
```